### DISCUSSION TITLE
Custom Dependencies

### DISCUSSION THREAD

**Discussion by pauldtill** [renovatebot/renovate#17638] [Request Help] [answered]
Hi,

In some of our pipelines, we use a yaml file (config.yaml) to define the required environment specific info. This file is held in a GitLab repository.

Inside this file, we set lines like this, which tells the pipeline where to get the shared code from and what version (we have multiple configs pointing at a single shared code repo);
`
DEPLOYCODEURL: git@gitlaburl.com:git-project-repo.git
`
`
CODEVERSION: 1.0.1 # git tag or branch
`

Using a regex manager we have configured renovate find these files (we can see it discovered in the logs as below), but it does nothing further with it. Is it possible to manage this sort of dependency using renovate?
`
DEBUG: Matched 1 file(s) for manager regex: config.yaml (repository=git-project-repo)
`

I've had a scan through the docs and can't find much in the way of examples that align with this requirement, but I'm sure I will have missed something. Or possiby we can just change our config file format slightly, so renovate recognises the dependency correctly.

Thanks,
Paul

**Comment by viceice** [comment]
yes, should be possible. check the debug logs.
if you need further help then you need to create a minimal reproduction

**Reply by pauldtill** [reply]
Thanks for the reply.

The debug logs show nothing much at all, it picks up the regex match on the file, but then never mention it again - see attached.
[debug.log](https://github.com/renovatebot/renovate/files/9491374/debug.log)
You can see 2 other deps that are valid, but nothing about the dependency inside the config.yaml file.

Is there any info on how/where I should provide a reproduction of this?
Is there somewhere specific in the docs that might help me a bit more on how to set things up to work for this?

**Comment by pauldtill** [comment, marked as answer]
I've got this working now with some modifications to my regex matching, I'd missed all of the necessary capture groups.
